src/sentry/api/endpoints/project_details.py

```diff
@@ -476,34 +504,7 @@ def get(self, request: Request, project: Project) -> Response:
             GlobalParams.ORG_SLUG,
             GlobalParams.PROJECT_SLUG,
         ],
-        request=inline_serializer(
-            "UpdateProject",
-            fields={
-                "slug": serializers.RegexField(
-                    DEFAULT_SLUG_PATTERN,
-                    help_text="Uniquely identifies a project and is used for the interface.",
-                    required=False,
-                    max_length=50,
-                    error_messages={"invalid": DEFAULT_SLUG_ERROR_MESSAGE},
-                ),
-                "name": serializers.CharField(
-                    help_text="The name for the project.",
-                    required=False,
-                    max_length=200,
-                ),
-                "platform": serializers.CharField(
-                    help_text="The platform for the project.",
-                    required=False,
-                    allow_null=True,
-                    allow_blank=True,
-                ),
-                "isBookmarked": serializers.BooleanField(
-                    help_text="Enables starring the project within the projects tab.",
-                    required=False,
-                ),
-                "options": serializers.DictField(help_text=_options_description, required=False),
-            },
-        ),
+        request=ProjectAdminSerializer,
```

Comment:
annoyingly, this orders the body params as
- child serializer params then
- parent serializer params
so `ProjectMemberSerializer` fields appear first before `ProjectAdminSerializer` fields. It should be the other way around.

@JoshFerge Would you have any idea how to fix this? We can't do it in the post-process hook b/c there's no way to tie body params to serializer once it's in JSON. I'm not familiar with the pre-process hook, but is it possible to do it there?

Reply:
why does order matter?

Reply:
ah I just think it's nice to have the fields show up in order of relevance/importance on the docs. Something like `slug` feels a lot more important to highlight than `isBookmarked`

Reply:
gotcha. I believe you could write a custom serializer extension https://drf-spectacular.readthedocs.io/en/latest/drf_spectacular.html#drf_spectacular.extensions.OpenApiSerializerExtension

it would have to override the https://github.com/tfranzel/drf-spectacular/blob/6e4180b7175e4fd07d09f1c9820c0aa40553f001/drf_spectacular/extensions.py#L50 map_serializer function here.

you would then register the extension and you could either have it run on all serializers or use the decorator to have it only run on this one. not sure of the utility of that tho

Reply:
I leave the order as is for now because the scope of what I'm doing right now is much smaller and more urgent. Good point to add to API improvements though for sure.

Reply:
Created a ticket in the backlog: https://getsentry.atlassian.net/browse/ER-1930